OlinkAnalyze/R/bridgeable_plot.R

```diff
@@ -0,0 +1,294 @@
+#'Plots for each bridgeable assays between two platforms / products
+#' Author : Amrita Kar
+#'
+#'Generates a combined plot per assay containing a violin and boxplot plot
+#'  for IQR ranges ;
+#'correlation plot of NPX values ;
+#'a median count bar plot and
+#'KS plots from 2 platforms
+#'
+#' @param data a data frame of results from bridge normalization generated by
+#'  olink_normalization()
+#' @param median_counts_threshold Numeric value to use for median counts
+#'  threshold for the two platforms. Default is 150
+#' @param min_count Numeric value to use for minimum counts
+#'  for the two platforms. Default is 10
+#'
+#' @return An object of class "ggplot", 4 combined plots for each protein.
+#' @export
+#' @examples
+#' \donttest{
+#'
+#'npx_ht <- OlinkAnalyze:::data_ht_small |>
+#'  dplyr::filter(SampleType == "SAMPLE") |>
+#'  dplyr::mutate(Project = "data1")
+#'
+#'npx_3072 <- OlinkAnalyze:::data_3k_small |>
+#'  dplyr::filter(SampleType == "SAMPLE") |>
+#'  dplyr::mutate(Project = "data2")
+#'
+#'overlapping_samples <- unique(intersect(npx_ht |>
+#'                                          dplyr::distinct(SampleID) |>
+#'                                          dplyr::pull(),
+#'                                        npx_3072 |>
+#'                                          dplyr::distinct(SampleID) |>
+#'                                          dplyr::pull()))
+#'
+#'data <- OlinkAnalyze::olink_normalization(df1 = npx_ht,
+#'                                          df2 = npx_3072,
+#'                                          overlapping_samples_df1 =
+#'                                            overlapping_samples,
+#'                                          df1_project_nr = "Explore HT",
+#'                                          df2_project_nr = "Explore 3072",
+#'                                          reference_project = "Explore HT")
+#'
+#'olinkids <- unique(paste0(data$OlinkID,"_",data$Assay))
+#'results <- bridgeable_plts(data = data,
+#'                           median_counts_threshold = 150,
+#'                           min_count = 10)
+#'names(results) <- olinkids
+#'}
+
+bridgeable_plts <- function(data,
+                            median_counts_threshold = 150,
+                            min_count = 10) {
+
+  if (!requireNamespace("ggpubr", quietly = TRUE)) {
+
+    stop("These plots require the ggpubr package.
+         Please install ggpubr before continuing.
+            install.packages(\"ggpubr\")")
+  }
+
+  out_plts <- list()
+
+  ids <- data |>
+    dplyr::select(OlinkID, .data[["BridgingRecommendation"]]) |> #HT OlinkID
+    dplyr::distinct() |>
```

Comment:
Not really...each OlinkID has 1 bridging recommendation. We need to add the bridging recommendation since we are plotting bridgeable and non bridgeable assays in different colored texts